internal/generate/scaffold.go

```diff
@@ -0,0 +1,95 @@
+package generate
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+//go:embed scaffold
+var scaffoldFS embed.FS
+
+type ScaffoldData struct {
+	AppName        string
+	DeploymentName string
+	Description    string
+	Environment    string
+	Namespace      string
+}
+
+func AppScaffold(appName string, environments []string, outputPath string) error {
+	for _, env := range environments {
+		files, err := generateAppScaffoldEnvironmentFiles(appName, env)
+		if err != nil {
+			return err
+		}
+
+		if err := files.Save(filepath.Join(outputPath, env)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func generateAppScaffoldEnvironmentFiles(appName, environment string) (*Files, error) {
+	tpl, err := template.New("tpl").ParseFS(scaffoldFS, "scaffold/*.yaml", "scaffold/**/*.yaml")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create template")
+	}
+
+	data := ScaffoldData{
+		AppName:        appName,
+		DeploymentName: fmt.Sprintf("%s-environment-%s", environment, appName),
+		Description:    fmt.Sprintf("%s example application", appName),
+		Environment:    environment,
+		Namespace:      environment,
+	}
+
+	files := &Files{}
+	err = fs.WalkDir(scaffoldFS, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return errors.Wrap(err, "error walking directory")
+		}
+
+		if d.IsDir() {
+			return nil
```

Comment:
Might want to make sure you want to "skip" directories rather than "return".

There's this: https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/io/fs/walk.go;l=12-15

Reply:
This seems to be incorrect - it skips the directory's content which, as I want the nested files, breaks the desired functionality.

Reply:
Ah gotcha, so you want to find the directory still, just not process the directory itself but the contents in it.

Reply:
Correct